### PR TITLE
XA assembler

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116137` at 92279097 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/91af68c52788b264f352c32e5d610393">dxa</a></li></li>
    <li><a href="https://gist.github.com/7ae6a3e12b0bb9f6cd292c4b01095cae">xa</a></li></li>
  </ul>
</details>

File: pkgs/development/compilers/xa/dxa.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  };
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  dontConfigure = true;
+
+  postPatch = ''
+    sed -i \
+        -e 's/^CC = gcc/CC = cc/' \
+        Makefile
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/xa/dxa.nix:23]: Can we replace this with substituteInPlace?

File: pkgs/development/compilers/xa/xa.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  dontConfigure = true;
+
+  postPatch = ''
+    sed -i \
+        -e 's/DESTDIR/PREFIX/' \
+        -e 's/^CC = gcc/CC = cc/' \
+        -e 's/^LDD = gcc/LDD = ld/' \
+        -e 's/^CFLAGS = -O2/CFLAGS ?=/' \
+        -e 's/^LDFLAGS = -lc/LDFLAGS ?= -lc/' \
+        Makefile
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/xa/xa.nix:24]: Can we replace this with substituteInPlace?

File: pkgs/development/compilers/xa/dxa.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/development/compilers/xa/dxa.nix]: YEP

**Pull request merged**